lib/pharos/phases/validate_host.rb

```diff
@@ -44,10 +44,10 @@ def check_role
       end
 
       def validate_unique_hostnames
-        duplicates = @config.hosts.reject { |h| h.address == @host.address }.select { |h| h.hostname == @host.hostname }
+        duplicates = @config.hosts.reject { |h| h.address == @host.address }.select { |h| h.hostname == @host.hostname && h.ssh_port == @host.ssh_port }
```

Comment:
Hmm... why are we calling this unique hostnames and then we validate unique addresses? 🤔

Reply:
`GatherFacts` does `host.hostname = transport.exec("hostname -f")`.

If it's a port forward to some other host, it should get a different hostname anyway. 

Maybe the unique hostname thing actually needs to just check for duplicate hostnames, no matter what the address / port is, so something like `(@config.hosts - [@host]).find { |h| h.hostname == @host.hostname }` ?

Reply:
Hmm, in fact, I think instead of reporting hostnames as duplicate, it should have been allowing duplicate hostnames for hosts with the same address?

`@config.hosts.reject { |h| h.address == @host.address }` would have been empty if there were two hosts with the same address but different ssh ports.

Reply:
We are mixing two different things here:
1. uniqueness of address:port which is used for ssh
2. uniqueness of hostname which is used to identify nodes in kubernetes

Reply:
config_schema will complain if addresses are equal and does not check ssh_port.

Needs to be fixed also.